### ISSUE TITLE
Fix `block-no-empty` false positives with the `css` function nested in a styled component

### ISSUE THREAD

**Issue by bradlocking** [stylelint/stylelint#6779]
### What steps are needed to reproduce the bug?

Lint the following styled components code in a `.tsx` file with the provided stylelint configuration and version. Other version numbers:
- `postcss-styled-syntax`: `^0.4.0`
- `stylelint-config-standard`: `^32.0.0`

Failing code snippet:
```tsx
const buttonHover = css`
    color: blue;
    text-decoration: none;
    border-bottom: 3px solid black;
`;

export const Button = styled.button`
    &:hover {
        ${buttonHover}
    }
`;
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
    "customSyntax": "postcss-styled-syntax",
    "files": ["*.ts", "*.tsx"],
    "extends": [
        "stylelint-config-standard"
    ],
    "rules": {}
}
```

### How did you run Stylelint?

CLI with `stylelint \"src/**/*.{ts,tsx}\"`

### Which version of Stylelint are you using?

`15.4.0`

### What did you expect to happen?

The CSS should be valid and pass the `block-no-empty` rule check.

### What actually happened?

`block-no-empty` Returns an error 

```bash
src/shared/components/Button/Button.styled.tsx
 159:41  ✖  Unexpected empty block  block-no-empty
```

### Does the bug relate to non-standard syntax?

Yes, it's related to CSS-in-JS (styled components)

**Comment by ybiquitous** [comment]
@bradlocking Thanks for the report with the reproduction code. I confirm the behavior.

Probably, this behavior is related to <https://github.com/hudochenkov/postcss-styled-syntax/issues/3>. Since the Stylelint built-in rules handle only the standard CSS, it seems hard to fix the rule.

As https://github.com/hudochenkov/postcss-styled-syntax/issues/3#issuecomment-1463351965 suggested, how about using the following workaround?

```
    &:hover {
        /* some comment ... */
        ${buttonHover}
    }
```

**Comment by bradlocking** [comment]
@ybiquitous - Thanks for the quick response. I actually did find this workaround myself, though opted to explicitly disable the rule with a comment alongside it until something more concrete exists, as a comment would be needed either way!

**Comment by ybiquitous** [comment]
I've tried out parsing the JSX code like this:

```js
// check.js
const { parse } = require('postcss-styled-syntax');

const code = `
styled.button\`
  &:hover {
    \${buttonHover}
  }
\`;
`;

console.dir(parse(code).nodes[0].nodes);
```

Running `node check.js` outputs the following. Notice `nodes: []`.

```
<ref *1> [
  Rule {
    raws: { before: '\n  ', between: ' ', after: '\n    ${buttonHover}\n  ' },
    type: 'rule',
    nodes: [],
    parent: Root {
      raws: [Object],
      type: 'root',
      nodes: [Circular *1],
      source: [Object],
      parent: [Document],
      [Symbol(isClean)]: false,
      [Symbol(my)]: true
    },
    source: { start: [Object], input: [Input], end: [Object] },
    selector: '&:hover',
    [Symbol(isClean)]: false,
    [Symbol(my)]: true
  }
]
```

The reason that the `block-no-empty` rule warns is `nodes` in the rule are empty. So, I think it may be hard to handle it on the Stylelint side.

@hudochenkov Do you have any thoughts about this?

**Comment by hudochenkov** [comment]
@ybiquitous it is easy to handle on Stylelint side, but I feel it might go against Stylelint vision.

In case like this issue author's there is no CSS equivalent to `${buttonHover}` as it's impossible to know what this could be. So `postcss-styled-syntax` puts this interpolation in `node.raws.after` (in our case `node` is a `&:hover` rule).

We might change `block-no-empty` to also check if `node.raws.after` is not empty. Should keep in mind that `raws.after` could have just spacing or new lines. See AST for standard CSS https://astexplorer.net/#/gist/f7cb8740646a1548bb4f765d9ef4b9c2/7ea111dec5e61c08e216c2d64a67ff2901b7723e I've put semicolon to show possible standard CSS, which probably is not valid though.

**Comment by ybiquitous** [comment]
@hudochenkov Thanks for the help! I've opened PR #6782 to fix such false positives since I think checking `node.raws.after` makes sense in the `block-no-empty` rule.

**Comment by bradlocking** [comment]
Thanks for your quick response to this!
